---
title: "R Notebook"
output: html_notebook
date: 06-08-2017
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir=normalizePath('../../'))
```


should use genome-wide data to redo it.

# Calculate gene load and methylation level


```{r}
library(data.table)
library(tidyr)
library(dplyr)
#library(cowplot)

# install.packages('RCurl', dependencies=TRUE, repos='http://cran.rstudio.com/')
#source("https://bioconductor.org/biocLite.R")
#biocLite("GenomicRanges")
library("GenomicRanges")


lg <- read.csv("cache/geneload_count1_homo1.csv")
head(lg)
lg1 <- subset(lg, taxa %in% "maize")
lg2 <- subset(lg, taxa %in% "teosinte")
lc <- merge(lg1[, c(1,3)], lg2[, c(1,3)], by="geneid")

cor.test(lc$wbar.x, lc$wbar.y)
```

```{r}
plot(lc$wbar.x, lc$wbar.y, xlab="maize", ylab="teosinte")
```

## plot load and methylation level

normalize the gene length
```{r}
######## format GFF and repeat files
gff <- fread("~/dbcenter/AGP/AGPv3/Zea_mays.AGPv3.25.gff3", header=FALSE, data.table=FALSE)
names(gff) <- c("seqname", "source", "feature", "start", "end", "score",
                    "strand", "frame", "attribute")
gen <- subset(gff, feature %in% "gene")
gen$id <- gsub("ID=gene:|;assembly_name=.*", "", gen$attribute)
gen <- select(gen, 1:5, id)
gen$length <- gen$end - gen$start
write.table(gen, "cache/agpv3_genelength.csv", sep=",", row.names=FALSE, quote=FALSE)
```


```{r}
len <- read.csv("cache/agpv3_genelength.csv")
res <- read.csv("cache/stat_exon_mean_var.csv")

res1 <- merge(res, lg1, by="geneid")
res1 <- merge(res1, len[, c("id", "length")], by.x="geneid", by.y="id")


plot(res1$mm, res1$wbar/res1$length, pch=16, cex=0.5, col="blue", 
     ylab="total homozygote load per gene", xlab="methylation level")
cor.test(res1$wbar/res1$length, res1$mm)




```
```{r}
res17 <- subset(res1, wbar/length > 0.003 & mm > 0.7)
cor.test(res17$wbar/res17$length, res17$mm)

```

Michelle's idea:
seperate out the evolutionary time
ask Michelle about it. I kinda of lot the idea about it.






