---
title: "COMET"
author: "Jinliang Yang"
date: "10/12/2016"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## COMETgazer

[COMETgazer methylation analysis software suite](https://github.com/rifathamoudi/COMETgazer/) consists of 3 software:

### COMETgazer: 
A bash script for methylome segmentation into COMETs based on OM values.
Part of the bash script is `blocks.cpp` (now in `~/bin/`) which is a `C++` program assigning CpGs to COMETs

Change `chrom in 1 10`. Copy the script and run in the sub-folder.

- Input : whole genome bisulfite sequencing methylation data that have been smoothed

Example:  

- Input file : `chr22.txt` as an example of Chromosome 22 only. The software loops through all 22 chromosomes 
- Output file : `chr22_blocks_verified.txt` example of COMET segmentation of Chromosome 22 only

### OORTcloud: 
A bash script for counting COMET distributions according to methylation level

- Input : this will be the output from COMETgazer e.g chr22_blocks_verified.txt. The software loops through all 22 chromosomes

Example:  
- Input file : `chr22_blocks_verified.txt` example of COMET segmentation of Chromosome 22 only. The software loops through all 22 chromosomes.   
- Output file : `low.txt` low methylation level COMET distribution genome wide `medium.txt` medium methylation level COMET distribution genome wide `high.txt` high methylation level COMET distribution genome wide

### COMETvintage: 
A template R script for DMC analysis using edgeR.

Example Input file : low.txt low methylation level COMET distribution genome wide for one methylome medium.txt medium methylation level COMET distribution genome wide for one methylome high.txt high methylation level COMET distribution genome wide for one methylome Example Output file : text file with the coordinates of DMCs
