---
author: "Jinliang Yang"
title: "Use the MCMCBC package to Scan Genic features"
date: 10-10-2017
output: html_notebook
---

## Setup using huskeR

```{r setup, echo=FALSE}
knitr::opts_chunk$set(echo = TRUE) 
knitr::opts_knit$set(root.dir=normalizePath('../../')) 
#library(tidyverse)

devtools::install_git(
  "https://git.unl.edu/jyang21/mcmcbc",
  credentials = git2r::cred_user_pass("jyang21", getPass::getPass())
)
ls(getNamespace("mcmcbc"), all.names=TRUE)
```

## Major function that will be used

```{r}
runmcmc_te_geneset <- function(JOBID, inputdf="largedata/type.csv")
    ## JOBID: control the array job number. [num, 1]
    ## inputdf: [data.frame, cols=comet_file:(comet blocks file, i.e. largedata/lcache/SFS_comet_blocks_CG.csv), 
    #                    length:(blocks quantile len, 0,1,2,3,4), TE:(yes, no), 
    #                    outRD:(chr, "out.RData"),
    #                    feature:(chr, "exon", "intron", "up1k", "gene"),
    #                    
    #                    optional:
    #                    geneset_file:(chr, csv, cols: geneid, value, i.e. "geneset.tex"),
    #                    gset:(chr, "below" or "above", "wholeset"),
    #                    cutoff:(num, num, 0.5)]
    ## geneset: gene set to determine which set to cal sfs. [data.frame, cols: geneid, value]
    ## cutoff: cutoff for the values to get up set and down set of the gene. [num, 0.5]
```



## Run for Type I TE

```{r}
## context
comet_file <- c("largedata/lcache/SFS_comet_blocks_CG.csv", "largedata/lcache/SFS_comet_blocks_CHG.csv")
## COMET length
ln <- 0:4
## feature
fea <- "Class I Retroelements"
## gene means 1kb upstream
TE <- "yes"

inputdf <- data.frame(comet_file=rep(comet_file, each=5), length=rep(ln, times=2), feature=fea, TE="yes")
inputdf$outRD <- paste0(gsub("csv", "", inputdf$comet_file), "_", "len",inputdf$length, "_typeI_TE", ".RData")

### TYPE I te scan with various comet length
write.table(inputdf, "largedata/mcmc_scan_input.csv", sep=",", row.names=FALSE, quote=FALSE)

library("huskeR")
run_Rcodes(inputdf, outdir="slurm-script", cmdno=1,
           rcodes = "profiling/5.MCMC_scan2/5.A.2_geneset_scan.R",
           arrayshid = "slurm-script/run_mcmc_te10.sh",
           email="yangjl0930@gmail.com", runinfo = c(FALSE, "jclarke", 2, "8G", "10:00:00"))


### RPKM mean
write.table(a, "largedata/gs_type.csv", sep=",", row.names=FALSE, quote=FALSE)
### RPKM variance
write.table(a, "largedata/rpkm_var_type.csv", sep=",", row.names=FALSE, quote=FALSE)




###>>> In this path: cd /home/jolyang/Documents/Github/methylation
###>>> RUN: sbatch -p bigmemm --mem 8G --ntasks=1 --exclude=bigmem1,bigmem3,bigmem6 --time 24:00:00 slurm-script/run_mcmc_gs10.sh


## COMET length
ln <- 0:4
## context
type <- c("CG", "CHG")
## gene means 1kb upstream
fs <- c("Class II/III Transposable Ele", "Class I Retroelements", "Other")

a <- data.frame(length=rep(ln, each=6), type=rep(type, each=3), fs=rep(fs, each=1))
a$id <- paste(a$type, a$length, a$fs, sep="_")
a$id <- gsub(" ", "", a$id)
a$TE <- "yes"

### RPKM mean
write.table(a, "largedata/te_type.csv", sep=",", row.names=FALSE, quote=FALSE)
library("farmeR")
run_Rcodes(inputdf=data.frame(file=1:30, out=1:30), outdir="slurm-script", cmdno=1,
           rcodes = "profiling/5.MCMC_scan/5.D.1_te-scan.R",
           arrayshid = "slurm-script/run_mcmc_te30.sh",
           email="yangjl0930@gmail.com", runinfo = c(FALSE, "bigmemm", 1, "8G"))


###>>> In this path: cd /home/jolyang/Documents/Github/methylation
###>>> RUN: sbatch -p jclarke --mem 8G --ntasks=2 --time 24:00:00 slurm-script/run_mcmc_te30.sh



```






